### ISSUE TITLE
Few peak overlaps between predicted TFBSs for TF1 and TF2

### ISSUE THREAD

**Issue by YichaoOU** [Jstacs/Jstacs#17]
Hello,

Thank you for this great tool. 

I have trained two models for TF1 and TF2 in cell line A and predicted TFBSs for the whole blood lineages, such as HSC, CMP, GMP, etc. In cell line A, I have 700+ co-binding peaks (TF1&TF2, IDR peaks). However, with the predicted TFBSs, I only have 10-100+ overlap peaks. I feel something is not right. 

Here is what I did:

(`-c`) Conserved peak is IDR peaks.

(`-r`) Relaxed peak is the union of MACS2 call-peaks

(`-a`) The accessibility data is single-end ATAC-seq bam file.

(`-m`) The motif data is the PWM for TF1 and TF2, and several others including (the given order is the same in training and prediction):

1. Ctcf_H1hesc_shift20_bdeu_order-20_comp1-model-1
2. intersect_all_relaxed_filtered_lslim3-model-2
3. intersect_all_relaxed_filtered_lslim3-model-5
4. intersect_all_relaxed_filtered_pwm-model-1
5. ENCSR000BHK_SP1-human_1_hg19-model-2
6. intersect_all_relaxed_filtered_lslim3-model-3
7. intersect_all_relaxed_filtered_lslim3-model-6
8. intersect_all_relaxed_filtered_lslim3-model-1
9. intersect_all_relaxed_filtered_lslim3-model-4
10. intersect_all_relaxed_filtered_lslim3-model-7

When I got `Predictions.tsv.gz`, I took all regions `>0.5` and used `bedtools merge` to merge the 50bp windows and output the final "called peaks". I then used these "called peaks" for TF1 and TF2 and took the intersection. The result is very few overlaps.

Is there any suggestion for me to better tune the model?

Thanks,
Yichao

**Comment by jgraux** [comment]
Regarding 3, the question is 
a) how many peaks do TF1 and TF2 have in the training data, respectively,
b) how many of those peaks in the training data overlap,
c) how many "peaks" do TF1 and TF2 have in the predictions, respectively,
d) how many of those "peaks" in the predictions overlap.

I would also assume that ATAC-seq data affect predictions a lot. However, considering your plot, the _shape_ of ATAC-seq coverage does not change too much. Internally, we do a mean normalization (basically, what MACS2 does with their dynamic Poisson parameter) for coverage, so shape should be the most important information. 
It might be interesting to inspect the ATAC-seq feature files (Chromatin_accessibility.tsv.gz) for one of the HSCs and MEG in this region to see, which features differ. Maybe, this could lead to a better normalization idea, or specific normalization for individual features.

**Comment by YichaoOU** [comment]
Hi @jgraux,

Thank you for the infomation. 

I'm wondering if you have the code to calculate auROC and auPRC given the prediction.tsv.gz and label.tsv.gz?

Thanks,
Yichao

**Comment by YichaoOU** [comment]
I got some updates. After many tunings, I still don't have a very good result, but your tool's prediction is definitely better than my baseline model which is only based on the two motifs occurrences and ATAC-seq peak overlap. 

For the ATAC-seq signal, I did check `Chromatin_accessibility.tsv.gz` and they are low for LTHSC and STHSC. So I used the S3norm to normalize the ATAC-seq data across different cell lines and provided the `bw` file for Catchitt. Now for the housekeeping gene, GAPDH, the atac-seq signal are similar in different cell line. The probability for the region I'm interested is increased, but still not passing 0.5. 

I found `-i`, the number of iterations, the smaller the value, the overall higher probability for each bin.

`-abb`, `-aba`, `-p`, not affect a lot, and seems that `-p` is not affecting the result at all. I tried 0.01, 0.05, 0.2, 0.5.

I also found that the input for training chromosomes and iteration chromosomes will affect output. I'm currently manipulating these, which I know we should not do it?

Finally, I found the model tends to learn positive correlation for motif and ATAC with the binding label (by that, I mean negative correlation, like missing a motif in an indicator for a peak, is not learnt by the model. May not be true, just my feeling). So, for TF2, which has more overlap with ATAC-seq peaks, the auPRC for leave-out chromosomes are ~0.3, whereas for TF1, auPRC is only about 0.15-0.2.

**Comment by YichaoOU** [comment]
Final updates for people who want to use Catchitt to predict TF co-binding.

1. Don't expect auPRC to be impressive. Remember for the best-predicted TF (i.e., CTCF, possibly because the motif is longer and binding is not changed significantly across cell type), the average auPRC for all group (leaderboard round 1) is 0.41. Even we are using the best algorithm, it doesn't mean we have tuned the best parameters and have the enough training data.
2. That being said, instead of reporting auPRC, we should come up with another evaluation metric, like some feature enrichment in the predicted peaks (I used the top 1k peak) vs random peaks, to impress reviewers. 
3. I found it is better to directly train the model with co-binding peaks, instead of training two models then taking the overlap. 
4. I tried multiple parameter combinations, results look similar (although #iter affects a lot), so choosing the default parameters makes the method justification easier.
